xdsl/xdsl_opt_main.py

```diff
@@ -213,7 +216,7 @@ def register_all_passes(self):
 
         Add other/additional passes by overloading this function.
         """
-        pass
+        self.available_passes['mpi-to-llvm'] = mpi_to_llvm_lowering
```

Comment:
I don't quite like this name. You are lowering to llvm + arith + memref + func, right? Maybe only call it `lower-mpi`?